pkg/controller/volume/expand/pvc_populator.go

```diff
@@ -107,6 +107,13 @@ func (populator *pvcPopulator) Sync() {
 			eventType := v1.EventTypeNormal
 			if err != nil {
 				eventType = v1.EventTypeWarning
+			} else {
+				// Do not log and record the event when we can not found the plugin
+				// and the StorageClassName is not set at the same time.
+				// See https://github.com/kubernetes/kubernetes/issues/77404
+				if len(pv.Spec.StorageClassName) == 0 {
+					return
+				}
```

Comment:
pvc_populator and expand_controller have too many similar codes and we may be able to move them out.